scripts/fetch_data.py

```diff
@@ -0,0 +1,274 @@
+
+    return rules
+
+
+def filter_by_period(target, dates, periods):
+    """Search through dictionary containing data for different 30 year periods,
+       find the desired period and return the target variable.
+    """
+    for key in periods.keys():
+        for date in dates:
+            if date in key:
+                try:
+                    return periods[key][target]
+                except KeyError as e:
+                    logger.warning('Bad target variable: {} error: {}'
+                                   .format(target, e))
+                    return None
+
+
+def get_nffd(fd, time, timescale):
+    """Given the number of frost days and a time period determine the number of
+       frost free days.
+    """
+    # TODO: Consider 360 day calendars here?
+    if fd is None:
+        return None
+
+    if timescale == 'yearly':
+        return 365 - fd
+    elif timescale == 'seasonal':
+        if time == '0':
+            return 89 - fd
+        elif time == '1' or time == '2':
+            return 92 - fd
+        elif time == '3':
+            return 91 - fd
+    else:
+        logger.warning(('Could not find matching time for time: {} and '
+                        'timescale: {}').format(time, timescale))
+        return None
+
+
+def query_backend(sesh, model, args):
+    """Return the desired variable for a particular climate model"""
+    if type(args['variable']) is dict:
+        tasmin = filter_by_period(args['spatial'], args['dates'],
+                                  multistats(sesh,
+                                             ensemble_name='p2a_files',
+                                             model=model,
+                                             emission=args['emission'],
+                                             time=args['time'],
+                                             area=args['area'],
+                                             variable=args['variable']['min'],
+                                             timescale=args['timescale'],
+                                             cell_method=args['cell_method']))
+        tasmax = filter_by_period(args['spatial'], args['dates'],
+                                  multistats(sesh,
+                                             ensemble_name='p2a_files',
+                                             model=model,
+                                             emission=args['emission'],
+                                             time=args['time'],
+                                             area=args['area'],
+                                             variable=args['variable']['max'],
+                                             timescale=args['timescale'],
+                                             cell_method=args['cell_method']))
+        try:
+            return mean([tasmin, tasmax])
+        except TypeError as e:
+            logger.debug(('Unable to get mean of tasmin: {} and tasmax: {} in '
```

Comment:
The reason that error occurs is that not every model has temperature data.  Therefore, when I query for it the result can be `None` and the `mean` operation wont work.  I still wanted there to be some kind of message because it may help us discover missing data but warning was getting very annoying during testing.  

I think what may be a good idea is to check the result of the query before trying to take the mean.  Then afterwards if the mean still doesn't work it can be an error.